public/js/surveyBuilderForm.js

```diff
@@ -0,0 +1,190 @@
+
+            $questionContainer.sortable(
+                {
+                    placeholder: "ui-state-highlight"
+                }
+            )
+
+            $('.remove-question').click(function()
+            {
+                $(this).parent('.new-question').remove()
+            })
+
+            //resetting form
+            $question.val('')
+            $type.val('text-input')
+            $required.prop('checked', false)
+            $typeOptions.remove()
+            addQuestionButton.prop("disabled",true)
+        }
+    )
+})
+
+
+//_________________________________FUNCTION DEFINITIONS__________________________________________
+
+
+/**
+ * Adds the facility to add and remove options for multiple choice questions. Initially this consists of a text
+ * field and an add option (+) button.
+ *
+ * @param $container OBJECT the created DOM objects are appended to the contents of this html element
+ */
+function addOptionsCreator($container)
+{
+    $container.append(
+        '<div id="question-options">' +
+        '<label for="option-text">Options:</label>' +
+        '<div id="new-option-container" class="input-group col-xs-2">' +
+        '<input type="text" id="option-text">' +
+        '<button class="btn input-group-addon" id="add-option">+</button>' +
+        '</div>' +
+        '</div>'
+    )
+
+    createAddOptionHandler()
+}
+
+
+/**
+ * Creates a handler for the click event on the add option button. The handler adds the option (providing it's
+ * non-empty) to the list of options for that question.
+ * Then validates the form.
+ */
+function createAddOptionHandler()
+{
+    $('#add-option').click(function()
+    {
+        if($('#option-text').val() !== '') // disables an empty string being added as an option
+        {
+            createOption($('#new-option-container'))
+        }
+
+        validateNewQuestion($('#question').val(), $('#add-question'), $('#input-selector').val())
+    })
+}
+
+
+/**
+ * Creates a disabled field containing the text passed in, with a remove (-) button .
+ *
+ * @param $optionContainer OBJECT the div containing the text input and add button for creating new options
+ */
+function  createOption($optionContainer)
+{
+    var $optionText = $optionContainer.find('input')
+    $optionContainer.before(
+        '<div class="input-group col-xs-2">' +
+        '<input class="bg-success" value="' + $optionText.val() + '" disabled>' +
+        '<button class="btn remove-option input-group-addon">-</button>' +
+        '</div>'
+    )
+    $optionText.val('')
+
+    $('.remove-option').click(function()
+    {
+        removeOption(this)
+    })
+
+}
+
+
+/**
+ * Removes the selected option from the list and then validates the form.
+ *
+ * @param currentOption OBJECT the remove button for the option to be removed
+ */
+function removeOption(currentOption)
+{
+    $(currentOption).parent('div').remove()
+    validateNewQuestion($('#question').val(), $('#add-question'), $('#input-selector').val())
```

Comment:
I had a look and could see no reason why they couldn't be variables, so started making the changes. I then remembered a chat I had with Mike yesterday about this and he said this is one of those development compromises: on one hand it's inefficient to keep making jQuery selections, but on the other validateNewQuestion() is called by other, unrelated, functions, so in order to enforce dependency injection the variables would have to be passed into places where it doesn't really make sense to. He concluded that it's better to leave as is. I'll go and undo the changes I started making then! lol